### PR TITLE
Fridge Events implemented

### PR DESCRIPTION
**Pull request by aisaiev**

### CODE REVIEW

File: ClassWork/Enums/FridgeState.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ClassWork.Enums
+{
+    public enum FridgeState
+    {
+        ON = 1,
```
Comment (alekseybeletsky) [inline ClassWork/Enums/FridgeState.cs]: Enum items should follow UpperCamelCase rule.

Comment (alekseybeletsky) [reply ClassWork/Enums/FridgeState.cs]: -1 to source code quality.

Comment (alekseybeletsky) [reply ClassWork/Enums/FridgeState.cs]: Why do you use reverse order?

File: ClassWork/Enums/MainDoorState.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ClassWork.Enums
+{
+    public enum MainDoorState
+    {
+        OPENED = 1,
```
Comment (alekseybeletsky) [inline ClassWork/Enums/MainDoorState.cs]: The same as for FridgeState.

File: ClassWork/Enums/SecondaryDoorState.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ClassWork.Enums
+{
+    public enum SecondaryDoorState
+    {
+        OPENED = 1,
```
Comment (alekseybeletsky) [inline ClassWork/Enums/SecondaryDoorState.cs]: The same as for FridgeState.

File: ClassWork/Enums/SecondaryDoorState.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ClassWork.Enums
+{
+    public enum SecondaryDoorState
```
Comment (alekseybeletsky) [inline ClassWork/Enums/SecondaryDoorState.cs]: Don't see a reason of using different but completely the same enums.
I'd use the only.

File: ClassWork/Fridge.cs [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        {
+            if (this.FridgeState == FridgeState.OFF)
+            {
+                this.FridgeState = FridgeState.ON;
+            }
+            else
+            {
+                this.FridgeState = FridgeState.OFF;
+            }
+            this.FridgeEvent?.Invoke(this, new FridgeEventArgs(this.GetFridgeState()));
```
Comment (alekseybeletsky) [inline ClassWork/Fridge.cs]: You implemented whole state for EventArgs in your custom FridgeEventArgs but don't use them at all.

File: ClassWork/Fridge.cs [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    {
+        public FridgeState FridgeState { get; private set; }
+
+        public MainDoorState MainDoorState { get; private set; }
+
+        public SecondaryDoorState SecondaryDoorState { get; private set; }
+
+        public event EventHandler<FridgeEventArgs> FridgeEvent;
+
+        public void OnChangeFridgeState()
```
Comment (alekseybeletsky) [inline ClassWork/Fridge.cs]: You should have implemented a list of public methods that allow User call them.

Comment (alekseybeletsky) [reply ClassWork/Fridge.cs]: -1 to Event agreement.

File: ClassWork/Fridge.cs (hunk 5 above)
Comment (alekseybeletsky) [reply ClassWork/Fridge.cs]: -2 to logic.

File: ClassWork/User.cs [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+namespace ClassWork
+{
+    public class User
+    {
+        private Fridge fridge;
+
+        public User(Fridge fridge)
+        {
+            this.fridge = fridge;
+            this.fridge.FridgeEvent += this.fridge.PrintFridgeState;
```
Comment (alekseybeletsky) [inline ClassWork/User.cs]: Fridge should not be subscribed to himself.
User should have a method that is responsible for handling the event.

Comment (alekseybeletsky) [reply ClassWork/User.cs]: -2 to logic.

File: ClassWork/Fridge.cs (hunk 6 above)
Comment (alekseybeletsky) [reply ClassWork/Fridge.cs]: And logic should invoke the event.

File: ClassWork/Enums/FridgeState.cs (hunk 1 above)
Comment (detores) [reply ClassWork/Enums/FridgeState.cs]: In the binary system 1 means is On and 0 is Off, so I used this logic.

**Pull request closed without merging**